core/src/main/java/org/commonjava/indy/core/content/DefaultContentManager.java

```diff
@@ -268,10 +269,38 @@ public Transfer retrieve( final ArtifactStore store, final String path, final Ev
         return item;
     }
 
+    private boolean checkMask( final ArtifactStore store, final String path )
```

Comment:
Adding a new decorator is troublesome since it could easily skip some code paths in default decorator. (it broke some ftests). As I mentioned in jira, we have to incorporate it to default decorator at this time (avoiding mistakes, less code, all ftests satisfied).